wandering-warriors/scripts/generate_gfx.py

```diff
@@ -0,0 +1,114 @@
+"""
+Generate full set of cuneiform graphics from just the 1 and 10 symbol
+"""
+from pathlib import Path
+from PIL import Image
+
+IMAGE_DIR = Path('.') / 'assets' / 'graphics'
+GENERATED_DIR = IMAGE_DIR / 'cuneiform'
+
+ONE = Image.open(IMAGE_DIR / 'cuneiform_one.png')
+TEN = Image.open(IMAGE_DIR / 'cuneiform_ten.png')
+
+
+# quick helper
+def blank_image_gen():
+    return Image.new('RGBA', (300, 300), (255, 255, 255, 0))
+
+# now finally start generating
+
+
+# blank
+blank_image_gen().save(GENERATED_DIR / 'c0.png')
+
+# one
+out = blank_image_gen()
+out.paste(ONE, (100, 100), mask=ONE)
+out.save(GENERATED_DIR / 'c1.png')
+
+# two
+out = blank_image_gen()
+[out.paste(ONE, (50 + 100 * i, 100), mask=ONE) for i in range(2)]
+out.save(GENERATED_DIR / 'c2.png')
+
+# three
+out = blank_image_gen()
+[out.paste(ONE, (100 * i, 100), mask=ONE) for i in range(3)]
+out.save(GENERATED_DIR / 'c3.png')
+
+# four
+out = blank_image_gen()
+[out.paste(ONE, (100 * i, 50), mask=ONE) for i in range(3)]
+out.paste(ONE, (100, 150), mask=ONE)
+out.save(GENERATED_DIR / 'c4.png')
+
+# five
+out = blank_image_gen()
+[out.paste(ONE, (100 * i, 50), mask=ONE) for i in range(3)]
+[out.paste(ONE, (50 + 100 * i, 150), mask=ONE) for i in range(2)]
+out.save(GENERATED_DIR / 'c5.png')
+
+# six
+out = blank_image_gen()
+for i in range(3):
+    for j in range(2):
+        out.paste(ONE, (100 * i, 100 * j + 50), mask=ONE)
+out.save(GENERATED_DIR / 'c6.png')
+
+# seven
+out = blank_image_gen()
+for i in range(3):
+    for j in range(3):
+        out.paste(ONE, (100 * i, 100 * j), mask=ONE)
+out.paste(ONE, (100, 200), mask=ONE)
+out.save(GENERATED_DIR / 'c7.png')
+
+# eight
+out = blank_image_gen()
+for i in range(3):
+    for j in range(3):
+        out.paste(ONE, (100 * i, 100 * j), mask=ONE)
+[out.paste(ONE, (50 + 100 * i, 200)) for i in range(2)]
+out.save(GENERATED_DIR / 'c8.png')
+
+# nine
+out = blank_image_gen()
+for i in range(3):
+    for j in range(3):
+        out.paste(ONE, (100 * i, 100 * j), mask=ONE)
+out.save(GENERATED_DIR / 'c9.png')
+
+# ten
+out = blank_image_gen()
+out.paste(TEN, (100, 100), mask=TEN)
+out.save(GENERATED_DIR / 'c10.png')
+
+# twenty
+out = blank_image_gen()
+out.paste(TEN, (25, 100), mask=TEN)
+out.paste(TEN, (150, 0), mask=TEN)
+out.save(GENERATED_DIR / 'c20.png')
+
+# thirty
+out = blank_image_gen()
+out.paste(TEN, (25, 100), mask=TEN)
+out.paste(TEN, (150, 0), mask=TEN)
+out.paste(TEN, (150, 200), mask=TEN)
+out.save(GENERATED_DIR / 'c30.png')
+
+# forty
+out = blank_image_gen()
+out.paste(TEN, (25, 100), mask=TEN)
+out.paste(TEN, (100, 50), mask=TEN)
+out.paste(TEN, (100, 150), mask=TEN)
+out.paste(TEN, (200, 0), mask=TEN)
+out.save(GENERATED_DIR / 'c40.png')
+
+# fifty
+out = blank_image_gen()
+out.paste(TEN, (25, 100), mask=TEN)
+out.paste(TEN, (100, 50), mask=TEN)
+out.paste(TEN, (100, 150), mask=TEN)
+out.paste(TEN, (200, 0), mask=TEN)
+out.paste(TEN, (200, 100), mask=TEN)
+out.save(GENERATED_DIR / 'c50.png')
```

Comment:
So this isn't necessary by any means, but if you wanted to structure this in a way that uses less repeated code you could use something like the following:
```py
paste_arg_sets = {
    # Can use functools.partial() if you want to preserve keyword args (such as *mask*)
    1: [(ONE, (100, 100), ONE)],
    2: [(ONE, (50 + 100 * i, 100), ONE) for i in range(2)],
    3: [(ONE, (100 * i, 100), ONE) for i in range(3)],
    4: [[(ONE, (100 * i, 50), ONE) for i in range(3)], 
        [(ONE, (100, 150), ONE)]],
    # ...
    # Nested list comp
    9: [[(ONE, (100 * i, 100 * j), ONE) for j in range(3)] for i in range(3)],
    # ...
    50: [
        (TEN, (25, 100), TEN),
        (TEN, (100, 50), TEN),
        (TEN, (100, 150), TEN),
        (TEN, (200, 0), TEN),
        (TEN, (200, 100), TEN),
    ],
}

def run_paste_commands(out, arg_sets):
    for args in arg_sets:
        if type(args) == list:
            run_paste_commands(out, args)
        else:
            # print(f"out.paste({args})")
            out.paste(args)

if __name__ == '__main__':
    for num, arg_sets in paste_arg_sets.items():
        out = blank_image_gen()
        run_paste_commands(out, arg_sets)
        out.save(GENERATED_DIR / f'c{num}.png')
```

Recursion is a powerful tool. (: